src/types/api-configuration-parameters.ts

```diff
@@ -0,0 +1,61 @@
+import { AxiosRequestConfig } from 'axios'
+
+export interface APIConfigurationParameters {
+  /**
+   * parameter for apiKey security
+   * @param name security name
+   * @memberof APIConfigurationParameters
+   */
+  apiKey?:
+    | string
+    | Promise<string>
+    | ((name: string) => string)
+    | ((name: string) => Promise<string>)
+  /**
+   * parameter for basic security
+   *
+   * @type {string}
+   * @memberof APIConfigurationParameters
+   */
+  username?: string
```

Comment:
I don't think there is a username or password in SP API. Where did you find that information?

Reply:
@moltar: They don't mention in the docs. But I found them in generated API model code.

Reply:
I think it might be just generator's assumptions / defaults. Check the API docs and the model definitions as the source of truth.

https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#connecting-to-the-selling-partner-api

You can also probably borrow some from Java SDK:

https://github.com/amzn/selling-partner-api-docs/blob/main/guides/developer-guide/SellingPartnerApiDeveloperGuide.md#connecting-to-the-selling-partner-api-using-a-generated-java-sdk

Reply:
I have removed `username`, `password` property and make `axios` an optional property on `APIConfigurationParameters`. Thanks.